### DISCUSSION TITLE
How to read ENV variables in kratos.yml?

### DISCUSSION THREAD

**Discussion by ulf1** [ory/kratos#1407] [Q&A] [answered]
A `kratos.yml` file is copied into the containers, and processed within the conatainers, e.g.
- https://github.com/ory/kratos/blob/master/quickstart.yml#L16
- https://github.com/ory/kratos/blob/master/quickstart.yml#L41

The `kratos.yml` contains UI URLs, e.g. 
```
...
  flows:
    error:
      ui_url: http://127.0.0.1:4455/error
...
```

How can I get an Env variable into these files?

The following doesn't work. 

```
# in my CLI
export PORT_OF_MY_APP=56789
docker compose -f mywebapp.yml -f manyotherthings -f thekratosauthapi.yml up --build
...
# in the thekratosauthapi.yml
services:
  kratos:
    environment:
      PORT_OF_MY_APP: ${PORT_OF_MY_APP}
...
# kratos.yml
  flows:
    error:
      ui_url: http://127.0.0.1:${PORT_OF_MY_APP}/error    # <- I guess Kratos' "serve" program doesn't do this ..
```

**Comment by Benehiko** [comment, marked as answer]
Hi @ulf1 

You can't inject environment variables into the kratos.yml. 
If you wanted to say set the port of your application through the environment variable e.g.
`serve.public.port: 4433` would become `export SERVE_PUBLIC_PORT=4433`. Environment variables override your configurations in `kratos.yml`.

You can find all of the kratos configurations and their corresponding environment variables [here](https://www.ory.sh/kratos/docs/reference/configuration)
